packages/cdktf/lib/testing/matchers.ts

```diff
@@ -15,6 +15,23 @@ export type SynthesizedStack = {
 };
 export type MatcherReturn = { message: () => string; pass: boolean };
 
+export interface IMatchers {
+  toHaveDataSourceWithProperties(
+    received: string,
+    resourceType: TerraformConstructor,
+    properties?: Record<string, any>
```

Comment:
I think I came across any issue related to and optional record type in jsii. Just using a string keyed map may be needed instead.